include/deal.II/base/index_set.h

```diff
@@ -1654,9 +1691,34 @@ IndexSet::add_indices(const ForwardIterator &begin, const ForwardIterator &end)
           ++q;
         }
 
-      add_range(begin_index, end_index);
+      tmp_ranges.emplace_back(begin_index, end_index);
```

Comment:
Not wrong, but would be easier to read if you said
```suggestion
      tmp_ranges.emplace_back({begin_index, end_index});
```

Reply:
We have a pair of two integers which we construct from the two arguments - it won't work if we do `emplace_back` with an initializer list (the compiler complains about that). It might work to do `push_back` with the initializer list (I have not tried), but in that case I would prefer `emplace_back`. So I suggest to leave as is.

Reply:
Good point!

Reply:
One could also do `emplace_back (std::make_pair(...,...))`. But it's really an entirely unimportant point. I was just a bit confused about what is actually emplaced: Two integers, or a pair.

Reply:
Wolfgang, the whole point of emplace_back() is to forward the arguments and construct the object in-place. Why would you construct the pair first? I think of the arguments of emplace_back() as being "pasted" as the constructor arguments.
I think Martin was doing the right thing from the start.

Reply:
> I think of construct+move as about as good as construct-in-place ;-)

If you are happy with construct+move, you might as well just use push_back() or am I missing something? Saving the move makes a lot of sense to me and is the whole reason for emplace_back.

Reply:
What I was *trying* to say that the performance implications are zero in case of a POD type as we have here. So one might opt for the code that's easier to read -- which I thought might be the one that makes it clear that we're pushing a pair, not two integers.

Of course, if the object being emplaced is more complex, there is no question about any of your arguments!